code/game/machinery/doors/airlock.dm

```diff
@@ -1218,3 +1195,46 @@ var/list/airlock_overlays = list()
 		secondsElectrified = 0
 		open()
 		safe = TRUE
+
+
+/obj/machinery/door/airlock/obj_break(damage_flag)
+	if(!(stat & BROKEN) && !(flags & NODECONSTRUCT))
+		stat |= BROKEN
+		if(!panel_open)
+			panel_open = 1
+		wires.cut_all()
+		update_icon()
```

Comment:
When it takes too much damage the airlock's wires get all cut, and the airlock is broken.